apps/dashboard/test/models/batch_connect/app_test.rb

```diff
@@ -250,4 +250,38 @@ def expected_clusters(*args)
       assert_equal expected_file, File.read("#{dir}/submit.yml")
     end
   end
+
+  test "form element is not an array" do
+    form_yml = <<~HEREDOC
+      ---
+      form:
+        should_have_been_an_array
+    HEREDOC
+
+    Dir.mktmpdir do |dir|
+      r = PathRouter.new(dir)
+      app = BatchConnect::App.new(router: r)
+      File.open("#{dir}/form.yml", 'w') { |file| file.write(form_yml) }
+      assert !app.valid?
+      assert_equal I18n.t('dashboard.batch_connect_invalid_form_array'), app.validation_reason 
+    end
+  end
+
+  test "attributes element is not a map" do
+    form_yml = <<~HEREDOC
+      ---
+      form:
+        - is_an_array
+      attributes:
+        should_be_a_map
```

Comment:
Trying to load an app in real life like this throws this error. So it seems like we need a test case around `build_session_context` too.

```
/var/www/ood/apps/sys/dashboard/app/models/batch_connect/app.rb:388:in `[]='
/var/www/ood/apps/sys/dashboard/app/models/batch_connect/app.rb:388:in `add_cluster_widget'
/var/www/ood/apps/sys/dashboard/app/models/batch_connect/app.rb:191:in `build_session_context'
/var/www/ood/apps/sys/dashboard/app/controllers/batch_connect/session_contexts_controller.rb:67:in `set_session_context'
/var/www/ood/apps/sys/dashboard/app/controllers/batch_connect/session_contexts_controller.rb:8:in `new'
```

Reply:
Tests added, but this PR actually fixes this by returning empty attributes if it's invalid.